stacker/actions/diff.py

```diff
@@ -0,0 +1,146 @@
+import logging
+
+from .. import exceptions
+from ..plan import COMPLETE, SKIPPED, Plan
+from . import build
+import difflib
+import json
+import sys
+
+logger = logging.getLogger(__name__)
+
+
+class Action(build.Action):
+    """ Responsible for diff'ing CF stacks in AWS and on disk
+
+    Generates the build plan based on stack dependencies (these dependencies
+    are determined automatically based on references to output values from
+    other stacks).
+
+    The plan is then used to pull the current CloudFormation template from
+    AWS and compare it to the generated templated based on the current
+    config.
+    """
+
+    def _diff_parameters(self, old_params, new_params):
```

Comment:
This logic is probably worth writing a test for.  It'll likely be easier to make it testable by breaking it out as a function.  You can see an example of that here:

https://github.com/remind101/stacker/blob/master/stacker/actions/build.py#L115